src/commands/generate_buildpack_matrix/command.rs

```diff
@@ -1,34 +1,70 @@
-use crate::buildpacks::{find_releasable_buildpacks, read_image_repository_metadata};
+use crate::buildpacks::{
+    find_releasable_buildpacks, read_buildpack_descriptor, read_image_repository_metadata,
+};
 use crate::commands::generate_buildpack_matrix::errors::Error;
+use crate::commands::resolve_path;
 use crate::github::actions;
 use clap::Parser;
-use libcnb_package::{
-    get_buildpack_target_dir, read_buildpack_data, BuildpackData, GenericMetadata,
+use libcnb_data::buildpack::{BuildpackDescriptor, BuildpackId};
+use libcnb_package::output::{
+    create_packaged_buildpack_dir_resolver, default_buildpack_directory_name,
 };
+use libcnb_package::CargoProfile;
 use std::collections::{BTreeMap, HashSet};
-use std::path::Path;
+use std::path::{Path, PathBuf};
 
 type Result<T> = std::result::Result<T, Error>;
 
 #[derive(Parser, Debug)]
 #[command(author, version, about = "Generates a JSON list of buildpack information for each buildpack detected", long_about = None)]
-pub(crate) struct GenerateBuildpackMatrixArgs;
+pub(crate) struct GenerateBuildpackMatrixArgs {
+    #[arg(long)]
+    pub(crate) working_dir: Option<PathBuf>,
+    #[arg(long)]
+    pub(crate) package_dir: PathBuf,
+    #[arg(long)]
+    pub(crate) release: Option<bool>,
+    #[arg(long, default_value = "x86_64-unknown-linux-musl")]
+    pub(crate) target: String,
+}
+
+pub(crate) fn execute(args: GenerateBuildpackMatrixArgs) -> Result<()> {
+    let working_dir = std::env::current_dir()
+        .map(|base| {
+            args.working_dir
+                .map_or(base.clone(), |path| resolve_path(&path, &base))
+        })
+        .map_err(Error::ResolveWorkingDir)?;
+
+    let package_dir = resolve_path(&args.package_dir, &working_dir);
+
+    let cargo_profile = if args.release.unwrap_or(true) {
```

Comment:
I think this is a bug? It should be `unwrap_or(false)` rather than `unwrap_or(true)`, since at the moment passing `--release` to the CLI gives a release build, but not passing it also gives a release build.

Either we should drop the `release` arg, and only support release builds here, or else default to non-release, and have `--release` as the way to specify release. 

I think my preference is just to drop the `release` are entirely -- this release automation is all about producing a final release product, not a development build.

Also, if we did want to support a `--release` arg, then we perhaps would instead want to change `args.release` from `Option<bool>` to `bool`, and instead set a Clap default value on the `GenerateBuildpackMatrixArgs` struct directly.

Reply:
Addressed in 4ff35503f2f9ef293637e482c7d047c3b4e77544